ci_framework/roles/polarion/tasks/main.yml

```diff
@@ -0,0 +1,123 @@
+      - cifmw_polarion_test_id is defined and cifmw_polarion_test_id != ""
+      - cifmw_polarion_update_testcases is defined and cifmw_polarion_update_testcases == "false" or cifmw_polarion_update_testcases == "true"
+      - cifmw_polarion_test_results_url is defined and cifmw_polarion_test_results_url != ""
+
+- name: Clone Jump repository
+  ansible.builtin.git:
+    repo: "{{ cifmw_polarion_jump_repo_url }}"
+    dest: "{{ cifmw_polarion_jump_repo_dir }}"
+    version: "{{ cifmw_polarion_jump_repo_branch }}"
+    force: true
+
+- name: Create virtualenv
+  ansible.builtin.pip:
+    name: pip
+    virtualenv: "{{ cifmw_polarion_jump_repo_dir }}/jump-venv"
+    virtualenv_command: "/usr/bin/python3 -m venv"
+    state: latest
+    extra_args: --upgrade
+  when:
+    - cifmw_polarion_jump_repo_dir is defined and cifmw_polarion_jump_repo_dir != ""
+
+# Will be removed before merge
+# - name: Temp task to use polarion-staging
+#   ansible.builtin.replace:
+#     path: "{{ cifmw_polarion_jump_repo_dir }}/{{ item }}"
+#     regexp: "{{ polarion_fqdn }}"
+#     replace: "{{ polarion_stage_fqdn }}"
+#   loop:
+#     - "helpers.py"
+#     - ".pylero"
+
+- name: Run prepare_pylero.sh
+  ansible.builtin.shell: >
+    source jump-venv/bin/activate && \ 
+    ./prepare_pylero.sh
+  args:
+    chdir: "{{ cifmw_polarion_jump_repo_dir }}"
+    executable: /bin/bash
+
+- name: Create {{ cifmw_polarion_jump_result_dir }} directory
+  ansible.builtin.file:
+    path: "{{ cifmw_polarion_jump_result_dir }}"
+    state: directory
+
+- name: Retrieve HTML content
+  ansible.builtin.uri:
+    url: "{{ cifmw_polarion_test_results_url }}"
+    return_content: yes
+  register: html_content
+
+- name: Extract XML URLs
+  ansible.builtin.set_fact:
+    xml_urls: "{{ html_content.content | regex_findall('<a href=\"\\.\\/([^\\/]+\\.xml)\">[^<]+<\\/a>') }}"
+
+- name: Download XML files
+  ansible.builtin.uri:
+    url: "{{ cifmw_polarion_test_results_url }}/{{ xml_url }}"
+    dest: "{{ cifmw_polarion_jump_result_dir }}/{{ xml_url | basename }}"
+  loop: "{{ xml_urls }}"
+  loop_control:
+    loop_var: xml_url
+  register: download_result
+  changed_when: download_result.status != 304
+  failed_when: false
+  until: download_result.status != 304
+  retries: 3
+  delay: 2
+
+- name: Find XML files in directory
+  ansible.builtin.find:
+    paths: "{{ cifmw_polarion_jump_result_dir }}"
+    patterns: "*.xml"
+    recurse: yes
+  register: xml_files
+
+- name: Validate XML files with xmllint
+  ansible.builtin.command: "xmllint --noout {{ item.path }}"
+  loop: "{{ xml_files.files }}"
+  register: xmllint_output
+  changed_when: false
+  failed_when: xmllint_output.rc != 0
+
+- name: Set XML files in a fact
+  set_fact:
+    found_xml_files: "{{ xml_files.files | map(attribute='path') | join(',') }}"
+
+- name: Update Test Runs
+  shell: >
+    source "{{ cifmw_polarion_jump_repo_dir }}/jump-venv/bin/activate" && \
+    python jump.py \
```

Comment:
you may use the python nested in `{{ cifmw_polarion_jump_repo_dir }}/jump-venv/bin/` directly, that would allow to NOT rely on a shell call